src/stores/global/AuthStore/AuthStore.ts

```diff
@@ -0,0 +1,222 @@
+            confirmed: false,
+            blocked: false,
+            createdAt: '',
+            updatedAt: ''
+        }
+    };
+    private _meta: Meta = Meta.initial;
+    private _initialization: boolean = false;
+    private _isAuthCheckComplete = false;
+
+    constructor() {
+        makeObservable<AuthStore, PRIVATE_FIELDS_AUTH>(this, {
+            _user: observable.ref,
+            _meta: observable,
+            _initialization: observable,
+            _isAuthCheckComplete: observable,
+            user: computed,
+            meta: computed,
+            initialization: computed,
+            isAuthCheckComplete: computed,
+            reset: action,
+            getAuth: action,
+            validateAuth: action,
+            initializeAuth: action
+        });
+    }
+
+    get user(): AuthDataModel {
+        return this._user;
+    }
+
+    get meta(): Meta {
+        return this._meta;
+    }
+
+    get initialization(): boolean {
+        return this._initialization;
+    }
+
+    get isAuthCheckComplete(): boolean {
+        return this._isAuthCheckComplete;
+    }
+
+    setToken(value: string) {
+        this._user.token = value;
+    }
+
+    async getAuth(
+        params: GetAuthParams
+    ): Promise<void> {
+        if (this._meta === Meta.loading) {
+            console.warn('Запрос уже выполняется');
+            return;
+        }
+
+        if (!params.email || !params.pass) {
+            this._meta = Meta.error;
+            return;
+        }
+
+
+        this._meta = Meta.loading;
+
+        try {
+            const response = await this._apiStore.request<AuthDataApi>({
+                method: HTTPMethod.POST,
+                data: {
+                    identifier: params.email,
+                    password: params.pass,
+                },
+                headers: {},
+                endpoint: AUTH_ENDPOINT
+            });
+
+            runInAction(() => {
+                if (response.success) {
+                    try {
+                        this._meta = Meta.success;
+                        this._user = normalizeAuthData(response.data);
+                        localStorage.setItem('token', this._user.token);
+                        this._initialization = true;
+                        this._isAuthCheckComplete = true;
+                    }
+                    catch (e) {
+                        console.log(e);
+                        this._meta = Meta.error;
+                        this._user = {
+                            token: '',
+                            user: {
+                                id: 0,
+                                username: '',
+                                email: '',
+                                provider: '',
+                                confirmed: false,
+                                blocked: false,
+                                createdAt: '',
+                                updatedAt: ''
+                            }
```

Comment:
громоздко и переписываем несколько раз

предлагаю в начале стора или в утилите рядом сделать:
```suggestion
 const setDefaultUser(): UserModel {
  return {
    id: 0,
    username: '',
    email: '',
    provider: '',
    confirmed: false,
    blocked: false,
    createdAt: '',
    updatedAt: ''
  };
}
```

Reply:
и потом просто ее вызывать